db/migrate/20190131170022_create_reports.rb

```diff
@@ -0,0 +1,17 @@
+class CreateReports < ActiveRecord::Migration[5.2]
+  def change
+    create_table :reports, id: :uuid, default: -> { "uuid_generate_v4()" } do |t|
+      t.uuid :partner_id
+      t.references :uploaded_by, index: true, foreign_key: { to_table: :publishers }, type: :uuid
+
+      t.string :amount_bat
+      t.datetime :created_at, null: false
```

Comment:
I generally prefer `t.timestamps`, but if this works with auto-update, then don't change.